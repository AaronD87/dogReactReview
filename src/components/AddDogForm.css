.form {
  margin: 3em 0 0.5em 0;
	color: #343434;
	font-weight: normal;
	font-size: 1.75em;
	line-height: 40px;
  font-family: 'Orienta', sans-serif;
  text-align: center;

}

.input {
  margin: 0em 1em;
  border: 0;
  outline: 0;
  background: transparent;
  border-bottom: 1px solid black;
  width: 7em;
  font-size: 1.1em;
}

.button {
  background-color: #0066A2;
  color: white;
  border-style: outset;
  border-color: #0066A2;
  height: 3em;
  width: 7em;
  font-weight: bold;
  font-family: 'Orienta', sans-serif;
  text-shadow: none;
  border-radius: 10%;
}

.button:focus {
  outline-width: 0;
}