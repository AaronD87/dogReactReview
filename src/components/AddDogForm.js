import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './AddDogForm.css';

class AddDogForm extends PureComponent {
  static propTypes = {
    addDog: PropTypes.func.isRequired
  };

  state = {
    name: '',
    age: '',
    weight: ''
  };
  
  handleSubmit = event => {
    event.preventDefault();
    const { name, age, weight } = this.state;
    this.props.addDog({ name, age: Number.parseInt(age), weight });
    
  };

  updateInput = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  render() {
    const { name, age, weight } = this.state;
    return (
      <>
        <form onSubmit={this.handleSubmit}>
          <span>Name: </span>
          <input type='text' name='name' value={name} onChange={this.updateInput} />
          <span>Age: </span>
          <input type='number' name='age' value={age} onChange={this.updateInput} />
          <span>Weight: </span>
          <input type='text' name='weight' value={weight} onChange={this.updateInput} />
          <button type='submit'>Add Dog</button>
        </form>
      </>
    );
  }
}

export default AddDogForm;
