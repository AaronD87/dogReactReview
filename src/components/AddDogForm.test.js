import React from 'react';
import { shallow } from 'enzyme';
import AddDogForm from './AddDogForm';


describe('<AddDogForm />', () => {
  it('renders', () => {
    const wrapper = shallow(<AddDogForm addDog={jest.fn()} />);
    
    expect(wrapper.exists()).toBe(true);
  });

  it('echos user text and form submit is called with user inputs', () => {
    const addDog = jest.fn();
    const wrapper = shallow(<AddDogForm addDog={addDog} />);

    wrapper.find('input').at(0).simulate('change', {
      target: { 
        value: 'Pete',
        name: 'name'   }
    });

    wrapper.find('input').at(1).simulate('change', {
      target: { 
        value: '12',
        name: 'age'   }
    });

    wrapper.find('input').at(2).simulate('change', {
      target: { 
        value: '70 lbs',
        name: 'weight'   }
    });

    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });

    expect(wrapper.state('name')).toEqual('Pete');
    expect(wrapper.state('age')).toEqual('12');
    expect(wrapper.state('weight')).toEqual('70 lbs');

    expect(addDog).toHaveBeenCalledWith({
      name: 'Pete',
      age: 12,
      weight: '70 lbs'
    });
  });

  it('matches a snapshot for AddDogForm', () => {
    const wrapper = shallow(<AddDogForm addDog={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
