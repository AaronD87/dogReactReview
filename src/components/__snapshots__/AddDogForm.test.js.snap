// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddDogForm /> matches a snapshot for AddDogForm 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <span>
      Name: 
    </span>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <span>
      Age: 
    </span>
    <input
      name="age"
      onChange={[Function]}
      type="number"
      value=""
    />
    <span>
      Weight: 
    </span>
    <input
      name="weight"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Dog
    </button>
  </form>
</Fragment>
`;
